NDTensors/src/extensions/gpuarrayscore.jl

```diff
@@ -0,0 +1,12 @@
+using .TypeParameterAccessors: TypeParameterAccessors, type_parameter
+using GPUArraysCore: AbstractGPUArray
+using Adapt: Adapt
+
+TypeParameterAccessors.default_type_parameters(::Type{<:AbstractGPUArray}) = (Float32, 1)
```

Comment:
I'm not sure if this is a good idea since I don't think there is any guarantee that subtypes will have that type parameter convention.

Reply:
That's a downside of the current `default_type_parameters`, it relies on type parameter positions instead of names, so it only works for concrete types, so we can't really make definitions for all subtypes of abstract types.

Reply:
Ahh yes this makes sense. So I should define this for `CuArrays` and `MetalArrays`?

Reply:
Yeah.

Reply:
Got it, I made a defaults for both of them separately. Thanks!